### ISSUE TITLE
Godot 4.0 support

### ISSUE THREAD

**Issue by mouring** [glass-brick/Scene-Manager#10]
This isn't so much a request, but a future note for when 4.0 comes you can pull back some work and not have to fight through the changes.  I've done a quick port in my tree to Godot 4.0.  In general the code should be mostly the same.  The following changes did occur:

- shaders now end with .gdshader and refuses to load the old 3.x named .shader
- assert() takes a static string so you can't do ("foo %s" % var) anymore.  I've just stripped the variable replacement for the time being
- yield -> await
- Fixed import/onready/tool statements
- is_abs_path is now is_absolute_path
- Moved button to non-programmable in example (mainly because I don't fully get callable and connect() feature)

**Comment by jabsatz** [comment]
Thanks for the heads up, will definitely check it out once Godot 4.0 releases

**Comment by jabsatz** [comment]
Hello @mouring, I've decided to open up a second branch in this repo called `godot-4.x`.

You're 100% welcome to open up a PR pointing to that branch. If everything looks good and it gets merged, I'll look into posting the separate branch to the Asset Lib.

For now we'll still be mostly working on the main branch, but my guess is once Godot 4 reaches beta or release status, we'll start actively maintaining and developing that branch.

And again, thank you so much for your work!

**Comment by mouring** [comment]
Never used Entity Singleton feature.  So never looked at it.  My testing was my application and the provided demos.

It will be a few days before I look at integrating your Godot 4.0apha branch into my main project.  If it works then I'll delete my fork.

**Comment by mouring** [comment]
It integrated well into my existing project.  I need to file a separate ticket as I extended the APIs as I needed greater control over non-scene loading fades.

**Comment by jabsatz** [comment]
Yes! If you need any new functionality please open a separate ticket and we'll talk about it there :)
